File: generator/filters/src/FixedRegion.cxx
#include "FixedRegion.h"

using namespace generator;

FixedRegion::FixedRegion(const std::string name):
  IMsgService(name),
  IAlgorithm(nullptr)
{
  declareProperty( "Eta"            , m_eta                 );
  declareProperty( "Phi"            , m_phi                 );
}



StatusCode FixedRegion::initialize()
{
  return StatusCode::SUCCESS;
}



StatusCode FixedRegion::execute( generator::Event& event )
{
  event.setEventNumber(-1);
  auto seed = generator::Seed(m_eta,m_phi);
  event.push_back(seed);
  return StatusCode::SUCCESS;
}



StatusCode FixedRegion::finalize()
{
  return StatusCode::SUCCESS;
}

File: generator/filters/python/FixedRegion.py
__all__ = ["FixedRegion"]

from GaugiKernel import Cpp
from GaugiKernel.macros import *
from ROOT import generator


class FixedRegion( Cpp ):

  def __init__( self, name,
                      Eta         : float=0.0,
                      Phi         : float=0.0,
                      OutputLevel : int=0 ): 
    
    Cpp.__init__(self, generator.FixedRegion(name))
    self.setProperty( "Eta"          , Eta           )
    self.setProperty( "Phi"          , Phi           )
    self.setProperty( "OutputLevel"  , OutputLevel   )



